insights/client/utilities.py

```diff
@@ -105,35 +113,84 @@ def write_to_disk(filename, delete=False, content=get_time()):
                 if err.errno != errno.ENOENT:
                     raise err
     else:
+        logger.debug("Writing '%s'" % filename)
         with open(filename, 'wb') as f:
             f.write(content.encode('utf-8'))
 
 
-def generate_machine_id(new=False,
-                        destination_file=constants.machine_id_file):
+def _get_rhsm_identity():
+    """Get the subscription-manager identity certificate UUID.
+
+    :returns: The subscription-manager UUID, or None if not found.
+    :rtype: str | None
     """
-    Generate a machine-id if /etc/insights-client/machine-id does not exist
+    if not HAS_RHSM:
+        return None
+
+    if hasattr(rhsm.config, "get_config_parser"):
+        # This function is only available since subscription-manager-1.27.1-1 (RHEL 8.3).
+        config = rhsm.config.get_config_parser()  # type: "rhsm.config.RhsmConfigParser"
+    elif hasattr(rhsm.config, "initConfig"):
+        config = rhsm.config.initConfig()  # type: "rhsm.config.RhsmConfigParser"
+    else:
+        logger.debug("")
+        return None
+
+    rhsm_certificate_dir_path = config.get("rhsm", "consumercertdir")  # type: str
+    rhsm_certificate_path = os.path.join(rhsm_certificate_dir_path, "cert.pem")  # type: str
+
+    if not os.path.isfile(rhsm_certificate_path):
+        return None
```

Comment:
maybe log here something too, eg to say that the system is most likely not registered?